python_modules/dagster/dagster/_daemon/sensor.py

```diff
@@ -724,16 +733,24 @@ def _evaluate_sensor(
 
         yield
     else:
-        yield from _handle_run_requests(
-            run_requests=sensor_runtime_data.run_requests,
-            cursor=sensor_runtime_data.cursor,
-            context=context,
-            instance=instance,
-            external_sensor=external_sensor,
-            workspace_process_context=workspace_process_context,
-            submit_threadpool_executor=submit_threadpool_executor,
-            sensor_debug_crash_flags=sensor_debug_crash_flags,
-        )
+        if sensor_runtime_data.run_requests:
+            yield from _handle_run_requests(
+                run_requests=sensor_runtime_data.run_requests,
+                cursor=sensor_runtime_data.cursor,
+                context=context,
+                instance=instance,
+                external_sensor=external_sensor,
+                workspace_process_context=workspace_process_context,
+                submit_threadpool_executor=submit_threadpool_executor,
+                sensor_debug_crash_flags=sensor_debug_crash_flags,
+            )
+
+        if sensor_runtime_data.backfill_requests:
+            _handle_backfill_requests(sensor_runtime_data.backfill_requests, instance, context)
```

Comment:
I think a worthy followup here would be to add more tick retry capabilities here (akin to the ones that we currently have for AMP ticks - more benefits of converging the daemons more) - we have previously been relying on requiring run keys to ensure that if there is a failure partway through and the sensor needs to retry, we can guarantee that things will stay idempotent. Since these backfill requests don't have run keys, we lose that escape hatch.

An alternative would be to allow users to apply a "backfill_key" here that is then tagged on the backfill and used to offer similar idempotency guarantees. Not clear that there would be a natural backfill key in the use cases motivating this though